src/openklant/components/contactgegevens/migrations/0004_alter_organisatie_adres_nummeraanduiding_id_and_more.py

```diff
@@ -0,0 +1,99 @@
+# Generated by Django 4.2.15 on 2025-01-09 09:22
+
+import json
+import logging
+import openklant.utils.validators
+from django.core.management import CommandError
+from django.db import migrations, models
+from django.db.models.functions import Length
+
+logger = logging.getLogger(__name__)
+
+
+def _log_records(records, model_name):
+    logger.info("--- START %s INFO ---", model_name)
+    logger.info(
+        "Found %s records for %s that have adres_nummeraanduiding_id length > 16 chars",
+        records.count(),
+        model_name,
+    )
+
+    for record in records:
+        logger.warning(
+            json.dumps(
+                {
+                    "model_name": model_name,
+                    "pk": record.pk,
+                    "uuid": str(record.uuid),
+                    "adres_nummeraanduiding_id": record.adres_nummeraanduiding_id,
+                }
+            )
+        )
+
+    logger.info("--- END %s INFO ---", model_name)
+
+
+def _check_field_length(apps, schema_editor):
+
+    Organisatie = apps.get_model("contactgegevens", "Organisatie")
+    Persoon = apps.get_model("contactgegevens", "Persoon")
+
+    organisatie_queryset = Organisatie.objects.annotate(
+        length=Length("adres_nummeraanduiding_id")
+    ).filter(length__gt=16)
```

Comment:
I think we should also include the entries that have length less than 16, because those wouldn't pass validation either